src/routes/login/callback/updateUser.ts

```diff
@@ -12,12 +12,14 @@ export async function updateUser(params: {
 	cookies: Cookies;
 }) {
 	const { userData, locals, cookies } = params;
+	// when using Google as OpenID provider, remember that it does not provide preferred_username
+	// example fix: userData.preferred_username = userData.email.split("@")[0];
```

Comment:
It's fine, the email will be displayed in the frontend if the username doesn't exist :)